src/NetAppFiles/NetAppFiles/Volume/GetNetAppFilesVolume.cs

```diff
@@ -0,0 +1,123 @@
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.NetAppFiles.Common;
+using Microsoft.Azure.Commands.NetAppFiles.Helpers;
+using Microsoft.Azure.Commands.NetAppFiles.Models;
+using Microsoft.Azure.Management.NetApp;
+using System.Linq;
+
+namespace Microsoft.Azure.Commands.NetAppFiles.Volume
+{
+    [Cmdlet(
+        "Get",
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetAppFilesVolume",
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = FieldsParameterSet), OutputType(typeof(PSNetAppFilesVolume))]
+    [Alias("Get-AnfVolume")]
+    public class GetAzureRmNetAppFilesVolume : AzureNetAppFilesCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The resource group of the ANF volume")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter()]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The name of the ANF account")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccounts",
+            nameof(ResourceGroupName))]
+        public string AccountName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The name of the ANF pool")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccounts/capacityPools",
+            nameof(ResourceGroupName),
+            nameof(AccountName))]
+        public string PoolName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = FieldsParameterSet,
+            HelpMessage = "The name of the ANF volume")]
+        [ValidateNotNullOrEmpty]
+        [Alias("VolumeName")]
+        [ResourceNameCompleter(
+            "Microsoft.NetApp/netAppAccounts/capacityPools/volumes",
+            nameof(ResourceGroupName),
+            nameof(AccountName),
+            nameof(PoolName))]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ResourceIdParameterSet,
+            HelpMessage = "The resource id of the ANF volume")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipeline = true,
+            ParameterSetName = ParentObjectParameterSet,
+            HelpMessage = "The pool object containing the volume to return")]
+        [ValidateNotNullOrEmpty]
+        public PSNetAppFilesPool PoolInputObject { get; set; }
```

Comment:
@leonardbf nit: I would rename this to just `PoolObject`

Reply:
Ok. Renamed.